@stellar/freighter-api/src/signMessage.ts

```diff
@@ -1,26 +1,48 @@
+import packageJson from "../package.json";
+import semver from "semver";
+import { Buffer } from "buffer";
+
 import { submitMessage } from "@shared/api/external";
 import { FreighterApiError } from "@shared/api/types";
 import { FreighterApiNodeError } from "@shared/api/helpers/extensionMessaging";
 import { isBrowser } from ".";
 
+type SignMessageV3Response = {
+  signedMessage: Buffer | null;
+  signerAddress: string;
+} & {
+  error?: FreighterApiError;
+};
+
+type SignMessageV4Response = {
+  signedMessage: string;
+  signerAddress: string;
+} & {
+  error?: FreighterApiError;
+};
+
 export const signMessage = async (
   message: string,
   opts?: {
     networkPassphrase?: string;
     address?: string;
   }
-): Promise<
-  { signedMessage: Buffer | null; signerAddress: string } & {
-    error?: FreighterApiError;
-  }
-> => {
+): Promise<SignMessageV3Response | SignMessageV4Response> => {
   if (isBrowser) {
     const req = await submitMessage(message, opts);
 
     if (req.error) {
       return { signedMessage: null, signerAddress: "", error: req.error };
     }
 
+    if (semver.gte(packageJson.version, "4.0.0") && req.signedMessage) {
```

Comment:
Why not hold off on merging this PR until we're ready to release 4.0, rather than have code thats version-aware?

Reply:
actually yeah on second thought, the code that should be version aware should be in the background script's message handler. We'll have 3.x and 4.x versions of the API installed on machines out there for some time so we do need to be backwards compatible here but I did it in the wrong place.

Reply:
ok I fixed this in fe25f4c68bc0194700837264bb97b629f0c9f554